plugins/module_utils/modules.py

```diff
@@ -115,6 +115,14 @@ def _is_changed(existing, payload):
         if v is not None:
             if k not in existing:
                 changed = True
+            elif isinstance(v, list):
+                # If the order of the list is different, it is considered as changed
+                if len(v) != len(existing[k]):
+                    changed = True
+                else:
```

Comment:
remove else we should check nested list as well.

Reply:
The Else condition provided in the loop below recursively checks if the existing attributes is equal to the payload (or the subset of the payload) , hence it is vital to the elif appended here.
Also , if the length of the existing attribute doesnt match the payload , there is no point of checking the list and we can directly send changed = True.